RecoTauTag/HLTProducers/interface/L1TJetsMatching.h

```diff
@@ -111,26 +118,141 @@ std::pair<std::vector<T>, std::vector<T>> categorise(
 
   return output;
 }
+template <typename T>
+std::tuple<std::vector<T>, std::vector<T>, std::vector<T>> L1TJetsMatching<T>::categoriseVBFPlus2CentralJets(
+    const std::vector<T>& pfMatchedJets, double pt1, double pt2, double pt3, double Mjj) const {  //60, 30, 50, 500
+  std::tuple<std::vector<T>, std::vector<T>, std::vector<T>> output;
+  unsigned int i1 = 0;
+  unsigned int i2 = 0;
+
+  double mjj = 0;
+  if (pfMatchedJets.size() > 1) {
+    for (unsigned int i = 0; i < pfMatchedJets.size() - 1; i++) {
+      const T& myJet1 = (pfMatchedJets)[i];
+
+      for (unsigned int j = i + 1; j < pfMatchedJets.size(); j++) {
+        const T& myJet2 = (pfMatchedJets)[j];
+
+        const double mjj_test = (myJet1.p4() + myJet2.p4()).M();
+
+        if (mjj_test > mjj) {
+          mjj = mjj_test;
+          i1 = i;
+          i2 = j;
+        }
+      }
+    }
+
+    const T& myJet1 = (pfMatchedJets)[i1];
+    const T& myJet2 = (pfMatchedJets)[i2];
+
+    std::vector<T> vec4jets;
+    vec4jets.reserve(4);
+    std::vector<T> vec5jets;
+    vec5jets.reserve(5);
+    std::vector<T> vec6jets;
+    vec6jets.reserve(6);
+    if (pfMatchedJets.size() > 3) {
+      if ((mjj > Mjj) && (myJet1.pt() >= pt3) && (myJet2.pt() > pt2)) {
+        vec4jets.push_back(myJet1);
+        vec4jets.push_back(myJet2);
+        unsigned int i3 = 0;
+        unsigned int i4 = 0;
+        bool filledi3 = false;
+        for (unsigned int i = 0; i < pfMatchedJets.size(); i++) {
+          if (i == i1)
+            continue;
+          if (i == i2)
+            continue;
+          if (!filledi3) {
+            i3 = i;
+            filledi3 = true;
+          } else {
+            i4 = i;
+            break;
+          }
+        }
+        const T& myJet3 = pfMatchedJets[i3];
+        const T& myJet4 = pfMatchedJets[i4];
+        vec4jets.push_back(myJet3);
+        vec4jets.push_back(myJet4);
+      }
+
+      if ((mjj > Mjj) && (myJet1.pt() < pt1) && (myJet1.pt() < pt3) && (myJet1.pt() > pt2) &&
+          (myJet2.pt() > pt2)) {  //60, 30, 50, 500
```

Comment:
`pt1>pt2` is enforced in the [loop evaluating mjj](https://github.com/cms-sw/cmssw/blob/ee4ca1a0e2e530e7a833eeb961cbf1e0ec98c0ef/RecoTauTag/HLTProducers/interface/L1TJetsMatching.h#L130) (since jets are pT-ordered), and all jets have `pt>pt2` (from [here](https://github.com/cms-sw/cmssw/blob/ee4ca1a0e2e530e7a833eeb961cbf1e0ec98c0ef/RecoTauTag/HLTProducers/interface/L1TJetsMatching.h#L276)). And if `myJet1.pt() > pt3`, then only 4 jets need to be matched so filling `vec5jets` and `vec6jets` is not needed (the events would be captured anyway by the "4 jet" version of the path/module. I hope this make sense

Reply:
> pt1>pt2 is enforced in the loop evaluating mjj (since jets are pT-ordered)

I would think this argument applies for `myJet1.pt() > myJet2.pt()`, but I was referring to the pT thresholds, which are ultimately the config parameters `"pt{1,2,3}Min"` (if the plugin assumes any hierarchy between those values, that should enforced in the constructor, to avoid mis-configurations).

> And if myJet1.pt() > pt3, then only 4 jets need to be matched so filling vec5jets and vec6jets is not needed (the events would be captured anyway by the "4 jet" version of the path/module). I hope this makes sense

I have the overall impression that the logic in `categorise*` could be made more obvious, but as long as the producer works as intended for all use cases, I won't insist on further changes (could be improved in a future PR).